### DISCUSSION TITLE
TCA and transitive dependencies ("error: no such module 'CustomDump'")

### DISCUSSION THREAD

**Discussion by protonjohn** [pointfreeco/swift-composable-architecture#2200] [Q&A]
Our app has an Xcode project, but we're beginning to modularize it out into smaller local Swift packages within the project. Some of these modules contain our UI. These modules depend on the TCA library. The TCA library depends on a number of other things, one of them being the `swiftui-navigation` framework, which in turn depends on `swift-custom-dump`. When trying to build the app, we end up getting the following compilation error from within the TCA project:

```
/Applications/Developer/DerivedData/ProtonVPN-bojknilkuvagaycpgobtatckypst/SourcePackages/checkouts/swiftui-navigation/Sources/_SwiftUINavigationState/AlertState.swift:1:8: error: no such module 'CustomDump'
import CustomDump
```

This error goes away when we include TCA as a dependency directly from the application, but re-appears occasionally when we move modules around or change the dependency structure (I'm guessing SPM only does a low-depth recursion when resolving dependencies). We thought about adding all of TCA's dependencies themselves as the app's dependencies to resolve the issue, but before forcing things through without thinking, we wanted to see if anybody else has experienced something similar, and if this disappointing limitation of SPM is a recent regression or a long-standing issue known to Apple.

**Comment by jaroslavas** [comment]
Temporary fix for this is not to use `prerelease/1.0` branch :(

**Reply by stephencelis** [reply]
I'd be a little surprised if `prerelease/1.0` was related to the issues @protonjohn is seeing, though a repro would be nice if you have one.

**Comment by stephencelis** [comment]
That's frustrating and a little confounding, since the error points at a file in a module that definitely depends on `CustomDump`. If you do think it's an SPM bug it's definitely file an issue with Swift!
